### PR TITLE
feat(TFD-152):  Use GCP properties optional in Input/OutputRuntime, a…

### PR DESCRIPTION
**Pull request by bchen-talend**
…nd try default application credential when no service account file provided

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Can't work with Default applicaiton credentials

**What is the new behavior?**
work with Default application credentials

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryConnection.java [hunk 1]
```diff
@@ -27,9 +27,12 @@
 public class BigQueryConnection {
 
     public static BigQuery createClient(BigQueryDatastoreProperties datastore) {
-        BigQuery bigquery = BigQueryOptions.newBuilder().setProjectId(datastore.projectName.getValue())
-                .setCredentials(createCredentials(datastore)).build().getService();
-        return bigquery;
+        if (datastore.serviceAccountFile.getValue() == null) {
```
Comment (pradasouvanlasy) [inline components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryConnection.java]: Seems safer to use `org.apache.commons.lang.StringUtils.isEmpty()`

File: components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryInputRuntime.java [hunk 2]
```diff
@@ -72,13 +71,15 @@ public ValidationResult initialize(RuntimeContainer container, BigQueryInputProp
...
 
-        GcpServiceAccountOptions gcpOptions = in.getPipeline().getOptions().as
-                (GcpServiceAccountOptions.class);
-        gcpOptions.setProject(datastore.projectName.getValue());
-        gcpOptions.setTempLocation(datastore.tempGsFolder.getValue());
-        gcpOptions.setCredentialFactoryClass(ServiceAccountCredentialFactory.class);
-        gcpOptions.setServiceAccountFile(datastore.serviceAccountFile.getValue());
-        gcpOptions.setGcpCredential(BigQueryConnection.createCredentials(datastore));
+        GcpServiceAccountOptions gcpOptions = in.getPipeline().getOptions().as(GcpServiceAccountOptions.class);
+        if (gcpOptions.getProject() == null) {
```
Comment (pradasouvanlasy) [inline components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryInputRuntime.java]: Seems safer to use `org.apache.commons.lang.StringUtils.isEmpty()`

Isn't there a way to know which Runner is used ?

File: components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryOutputRuntime.java [hunk 3]
```diff
@@ -61,13 +60,15 @@ public PDone expand(PCollection<IndexedRecord> in) {
...
 
-        GcpServiceAccountOptions gcpOptions = in.getPipeline().getOptions().as
-                (GcpServiceAccountOptions.class);
-        gcpOptions.setProject(datastore.projectName.getValue());
-        gcpOptions.setTempLocation(datastore.tempGsFolder.getValue());
-        gcpOptions.setCredentialFactoryClass(ServiceAccountCredentialFactory.class);
-        gcpOptions.setServiceAccountFile(datastore.serviceAccountFile.getValue());
-        gcpOptions.setGcpCredential(BigQueryConnection.createCredentials(datastore));
+        GcpServiceAccountOptions gcpOptions = in.getPipeline().getOptions().as(GcpServiceAccountOptions.class);
+        if (gcpOptions.getProject() == null) {
```
Comment (pradasouvanlasy) [inline components/components-bigquery/bigquery-runtime/src/main/java/org/talend/components/bigquery/runtime/BigQueryOutputRuntime.java]: Seems safer to use `org.apache.commons.lang.StringUtils.isEmpty()`

Isn't there a way to know which Runner is used ?

**Comment by RyanSkraba** [comment]
LGTM, after verifying the expected values for "empty" properties (null or "").

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/258/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/258/)

**Pull request merged**